add-on/src/lib/ipfs-companion.js

```diff
@@ -43,9 +46,14 @@ module.exports = async function init () {
       onCopyAddressAtPublicGw: () => copier.copyAddressAtPublicGw()
     })
     modifyRequest = createRequestModifier(getState, dnsLink, ipfsPathValidator)
+    ipfsProxy = createIpfsProxy(() => ipfs, getState)
     registerListeners()
     await setApiStatusUpdateInterval(options.ipfsApiPollMs)
-    await storeMissingOptions(options, optionDefaults, browser.storage.local)
+    await storeMissingOptions(
+      await browser.storage.local.get(),
```

Comment:
Can you explain this change? The code looks fine, just curious.

Reply:
Yes! Previously `storeMissingOptions` wasn't doing anything.

**On first run**, `options` will be the same as `optionDefaults`. Due to the line earlier:

```js
const options = await browser.storage.local.get(optionDefaults)
```

Regarding the parameter to `get`:

> A key (string) or keys (an array of strings or an object specifying default values) to identify the item(s) to be retrieved from storage
> https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/StorageArea/get#Parameters

...so, it finds `{}` in storage and then assigns `optionDefaults` to it.

So `storeMissingOptions` is given two identical objects and does nothing.

Similarly for subsequent runs, options will be assigned any missing defaults so there's never anything to store!

Instead, we get a fresh copy of the current options, without defaults applied.

Reply:
Nice catch!  I think I've made this bug while porting from legacy SDK to WebExtension APIs. 
Due to the "fallback to defaults" it worked anyway, so we never noticed :):)

Reply:
...probably should be in a separate PR - apologies